cmd/mdatagen/metrics_v2.tmpl

```diff
@@ -198,11 +202,39 @@ func (mb *MetricsBuilder) Emit(ro ...ResourceOption) pmetric.Metrics {
 {{ range $name, $metric := .Metrics -}}
 // Record{{ $name.Render }}DataPoint adds a data point to {{ $name }} metric.
 func (mb *MetricsBuilder) Record{{ $name.Render }}DataPoint(ts pcommon.Timestamp
-	{{- if $metric.Data.HasMetricValueType }}, val {{ $metric.Data.MetricValueType.BasicType }}{{ end }}
-	{{- range $metric.Attributes -}} , {{ .RenderUnexported }}AttributeValue string{{ end }}) {
+	{{- if $metric.Data.HasMetricInputType }}, val {{ $metric.Data.MetricInputType.String }}
+	{{- else }}
+	{{- if $metric.Data.HasMetricValueType }}, val {{ $metric.Data.MetricValueType.BasicType }}{{- end }}
+	{{- end -}}
+	{{- range $metric.Attributes -}} , {{ .RenderUnexported }}AttributeValue string{{ end }})
+	{{- if $metric.Data.HasMetricInputType }} error{{ end }} {
+	{{- if $metric.Data.HasMetricInputType }}
```

Comment:
```suggestion
	{{- if $metric.Data.HasMetricInputType }} error{{ end }} {
	{{- if $metric.Data.HasMetricInputType }}
```
Should these be combined?

Reply:
I couldn't think of a very clean way to do that with the single `{` at the end of the line there that's needed either way for the end of the function declaration